tests/integ/modin/types/test_timedelta_indexing.py

```diff
@@ -389,3 +389,183 @@ def loc_enlargement(key, item, df):
+    "key, api, query_count, join_count",
+    [
+        [2, "iat", 1, 2],
+        [native_pd.Timedelta("1 days 1 hour"), "at", 2, 2],
+        [[2, 1], "iloc", 1, 2],
+        [
+            [
+                native_pd.Timedelta("1 days 1 hour"),
+                native_pd.Timedelta("1 days 1 hour"),
+            ],
+            "loc",
+            1,
+            1,
+        ],
+        [slice(1, None), "iloc", 1, 0],
+        [[True, False, False, True], "iloc", 1, 1],
+        [[True, False, False, True], "loc", 1, 1],
+    ],
+)
+def test_series_with_timedelta_index(key, api, query_count, join_count):
+    td_idx = native_pd.TimedeltaIndex(
+        [
+            native_pd.Timedelta("1 days 1 hour"),
+            native_pd.Timedelta("2 days 1 minute"),
+            native_pd.Timedelta("3 days 1 nanoseconds"),
+            native_pd.Timedelta("100 nanoseconds"),
+        ]
+    )
+    snow_td_idx = pd.TimedeltaIndex(td_idx)
+
+    data = [1, 2, 3, 4]
+    native_series = native_pd.Series(data, index=td_idx)
+    snow_series = pd.Series(data, index=snow_td_idx)
+
+    with SqlCounter(query_count=query_count, join_count=join_count):
+        if is_scalar(key):
+            assert getattr(snow_series, api)[key] == getattr(native_series, api)[key]
+        else:
+            eval_snowpark_pandas_result(
+                snow_series, native_series, lambda s: getattr(s, api)[key]
+            )
+
+
+@pytest.mark.parametrize(
+    "key, api, query_count, join_count",
+    [
+        [2, "iat", 1, 2],
+        [native_pd.Timedelta("1 days 1 hour"), "at", 2, 2],
+        [[2, 1], "iloc", 1, 2],
+        [
+            [
+                native_pd.Timedelta("1 days 1 hour"),
+                native_pd.Timedelta("1 days 1 hour"),
+            ],
+            "loc",
+            1,
+            1,
+        ],
+        [slice(1, None), "iloc", 1, 0],
+        [[True, False, False, True], "iloc", 1, 1],
+        [[True, False, False, True], "loc", 1, 1],
+    ],
+)
+def test_df_with_timedelta_index(key, api, query_count, join_count):
+    td_idx = native_pd.TimedeltaIndex(
+        [
+            native_pd.Timedelta("1 days 1 hour"),
+            native_pd.Timedelta("2 days 1 minute"),
+            native_pd.Timedelta("3 days 1 nanoseconds"),
+            native_pd.Timedelta("100 nanoseconds"),
+        ]
+    )
+    snow_td_idx = pd.TimedeltaIndex(td_idx)
+
+    data = [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
+    native_df = native_pd.DataFrame(data, index=td_idx)
+    snow_df = pd.DataFrame(data, index=snow_td_idx)
+
+    with SqlCounter(query_count=query_count, join_count=join_count):
+        if is_scalar(key):
+            assert getattr(snow_df, api)[key, 0] == getattr(native_df, api)[key, 0]
+        else:
+            eval_snowpark_pandas_result(
+                snow_df, native_df, lambda s: getattr(s, api)[key]
+            )
+
+
+def test_df_with_timedelta_index_enlargement_during_indexing():
```

Comment:
would it be possible for this test to be parametrized, rather than a sequence of SqlCounter blocks?

Reply:
I think it would become more dense since some of the tests are run on `snow_df["a"]`. If you still prefer that I parameterize the test I can try to do that